File: Algorithm/leetcode_222.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_222.py
# Author            : Andy
# Date              : 2022.09.24
# Last Modified Date: 2022.09.24
# Last Modified By  : Andy


class Solution:

    def countNodes(self, root):
        count = 0
        if root is None:
            return count

        def dfs(node):
            nonlocal count
            if node is None:
                return

            if node is not None:
                count += 1

                dfs(node.left)
                dfs(node.right)

        dfs(root)
        return count

File: Algorithm/leetcode_222.go
/**
 * File              : leetcode_222.go
 * Author            : Andy
 * Date              : 2022.09.24
 * Last Modified Date: 2022.09.24
 * Last Modified By  : Andy
 */

package countNodes

func countNodes(root *TreeNode) int {
	count := 0
	var dfs func(node *TreeNode, count *int)
	dfs = func(node *TreeNode, count *int) {
		if node == nil {
			return
		} else {
			*count++
			dfs(node.Left, count)
			dfs(node.Right, count)
		}
	}
	dfs(root, &count)
	return count
}
